x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -138,56 +139,50 @@ export function mapDestroyed() {
 }
 
 export function mapExtentChanged(mapExtentState: MapExtentState) {
-  return async (
+  return (
     dispatch: ThunkDispatch<MapStoreState, void, AnyAction>,
     getState: () => MapStoreState
   ) => {
-    const dataFilters = getDataFilters(getState());
-    const { extent, zoom: newZoom } = mapExtentState;
-    const { buffer, zoom: currentZoom } = dataFilters;
-
-    if (extent) {
-      let doesBufferContainExtent = false;
-      if (buffer) {
-        const bufferGeometry = turfBboxPolygon([
-          buffer.minLon,
-          buffer.minLat,
-          buffer.maxLon,
-          buffer.maxLat,
-        ]);
-        const extentGeometry = turfBboxPolygon([
-          extent.minLon,
-          extent.minLat,
-          extent.maxLon,
-          extent.maxLat,
-        ]);
-
-        doesBufferContainExtent = turfBooleanContains(bufferGeometry, extentGeometry);
-      }
-
-      if (!doesBufferContainExtent || currentZoom !== newZoom) {
-        // snap to the smallest tile-bounds, to avoid jitter in the bounds
-        dataFilters.buffer = expandToTileBoundaries(extent, Math.ceil(newZoom));
-      }
+    const { extent, zoom: nextZoom } = mapExtentState;
+    const { buffer: prevBuffer, zoom: prevZoom } = getDataFilters(getState());
+
+    let doesPrevBufferContainNextExtent = true;
```

Comment:
Changed `doesPrevBufferContainNextExtent` default value from `false` to `true` because `false` was hiding an edge case. In the case where prevBuffer is undefined, one should be generated. This explicit case is now checked when setting buffer in dispatch of MAP_EXTENT_CHANGED action